component/componenttest/configtest.go

```diff
@@ -13,7 +13,7 @@ import (
 )
 
 // The regular expression for valid config field tag.
-var configFieldTagRegExp = regexp.MustCompile("^[a-z0-9][a-z0-9_]*$")
+var configFieldTagRegExp = regexp.MustCompile("^([a-z0-9][a-z0-9_]*)?$")
```

Comment:
I had to make this update to address tags like the one in the test below (`mapstructure:",remain"`), which was added by the new version of the Go SDK config package. No existing tests fail, but let me know if this has the potential to cause issues.

Reply:
I added a changelog for this change since it impacts behavior for API consumers. We can also make this change in a separate PR if desired.